### ISSUE TITLE
Some '...Details()' functions cause EDDI error if passed empty data.

### ISSUE THREAD

**Issue by Darkcyde13** [EDCD/EDDI#1623]
## EDDI version in which issue is found
V3.5.1

## VoiceAttack version in which issue is found (as applicable)
N/A

## Steps to reproduce
Run the following '...Details()' functions with empty data, i.e. '{CargoDetails("")}'
CargoDetails
MaterialDetails
RouteDetails

## Expected
Nothing to be spoken.

## Observed
EDDI reports "There is a problem with the (whatever) script.  Object reference not set to an instance of an object".

## Investigation
I was looking through the default scripts when I came to 'Material inventory report', and when I clicked the test button, I got the above error message.

I narrowed it down to the lines setting material to MaterialDetails().  I had run the test while Elite was not running, and thus the state variables had not been set.  It seems that EDDI still considered them to exist, but empty, as in just "".

I then went through all '...Details()' functions, testing them by simply running {...Details("")}.  All functions were silent, apart from the three; Cargo, Material, and Route, which all produce the error.  Is it possible they are all missing some kind of error trap that the others have?

I admit, I have made the assumption that the Details() functions all operate in a similar fashion, and I understand that these three may just work in a different way.  Still, I thought I'd bring it to your attention, as the test button in 'Material inventory report' gives the error if the game isn't running (or hasn't been run while EDDI is open?).

EDIT:  I also get the same error when clicking the test button on 'Material location report' and 'Material required report'.

**Comment by Darkcyde13** [comment]
Having moved on to the next few scripts, I've found that 'Material use report' is very similar to those reports mentioned above, but has additional checks to prevent the error being reported:
```
{if !material && len(state.eddi_context_material_name) > 0:
    {_ Fetch from context _}
    {set material to MaterialDetails(state.eddi_context_material_name)}
}

{if material && material.name: …. 
```
Adding these lines to the reports also fixes them.  It's mainly the `&& len(state.eddi_context_material_name) > 0:` that stops the error.  So I've now added these into my personality.
